In [1]:
from os.path import exists
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Student Perfomance Dataset

- Is hosted the [UC Irvine Machine Learning Repository](https://archive-beta.ics.uci.edu/ml/datasets/student+performance)
- [Paulo Cortez](http://www3.dsi.uminho.pt/pcortez) from Universidade do Minho (Portugal)
- The data was collected from two public schools in portugal.
- Conditions:
    - Add the citation "P. Cortez and A. Silva. Using Data Mining to Predict Secondary School Student Performance. In A. Brito and J. Teixeira Eds., Proceedings of 5th FUture BUsiness TEChnology Conference (FUBUTEC 2008) pp. 5-12, Porto, Portugal, April, 2008, EUROSIS, ISBN 978-9077381-39-7."
    - This dataset is licensed under a [Creative Commons Attribution 4.0 International (CC BY 4.0) license](https://creativecommons.org/licenses/by/4.0/). This allows for the sharing and adaptation of the datasets for any purpose, provided that the appropriate credit is given. (See above)



Let's download it 

if not exists("student.zip"):
    !wget https://archive.ics.uci.edu/ml/machine-learning-databases/00320/student.zip

and unzip it

!unzip student.zip

Quick inspection

!cat student.txt

Ok let's look at that R file:

!cat student-merge.R

This is a merge on common columns. We can load both files `student-mat.csv` and `student-por.csv` using pandas.

In [8]:
df1 = pd.read_csv("student-data/student-mat.csv", sep=";")
df2 = pd.read_csv("student-data/student-por.csv", sep=";")

Let see their columns:

In [9]:
df1.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

In [10]:
df2.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

In [11]:
df1.describe()

age        Medu        Fedu  traveltime   studytime    failures  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean    16.696203    2.749367    2.521519    1.448101    2.035443    0.334177   
std      1.276043    1.094735    1.088201    0.697505    0.839240    0.743651   
min     15.000000    0.000000    0.000000    1.000000    1.000000    0.000000   
25%     16.000000    2.000000    2.000000    1.000000    1.000000    0.000000   
50%     17.000000    3.000000    2.000000    1.000000    2.000000    0.000000   
75%     18.000000    4.000000    3.000000    2.000000    2.000000    0.000000   
max     22.000000    4.000000    4.000000    4.000000    4.000000    3.000000   

           famrel    freetime       goout        Dalc        Walc      health  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean     3.944304    3.235443    3.108861    1.481013    2.291139    3.554430   
std      0.896659    0.998862    1.113278    0.890741    1.287897    1.390303   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    3.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

         absences          G1          G2          G3  
count  395.000000  395.000000  395.000000  395.000000  
mean     5.708861   10.908861   10.713924   10.415190  
std      8.003096    3.319195    3.761505    4.581443  
min      0.000000    3.000000    0.000000    0.000000  
25%      0.000000    8.000000    9.000000    8.000000  
50%      4.000000   11.000000   11.000000   11.000000  
75%      8.000000   13.000000   13.000000   14.000000  
max     75.000000   19.000000   19.000000   20.000000

In [12]:
df2.describe()

age        Medu        Fedu  traveltime   studytime    failures  \
count  649.000000  649.000000  649.000000  649.000000  649.000000  649.000000   
mean    16.744222    2.514638    2.306626    1.568567    1.930663    0.221880   
std      1.218138    1.134552    1.099931    0.748660    0.829510    0.593235   
min     15.000000    0.000000    0.000000    1.000000    1.000000    0.000000   
25%     16.000000    2.000000    1.000000    1.000000    1.000000    0.000000   
50%     17.000000    2.000000    2.000000    1.000000    2.000000    0.000000   
75%     18.000000    4.000000    3.000000    2.000000    2.000000    0.000000   
max     22.000000    4.000000    4.000000    4.000000    4.000000    3.000000   

           famrel    freetime       goout        Dalc        Walc      health  \
count  649.000000  649.000000  649.000000  649.000000  649.000000  649.000000   
mean     3.930663    3.180277    3.184900    1.502311    2.280431    3.536210   
std      0.955717    1.051093    1.175766    0.924834    1.284380    1.446259   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    2.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

         absences          G1          G2          G3  
count  649.000000  649.000000  649.000000  649.000000  
mean     3.659476   11.399076   11.570108   11.906009  
std      4.640759    2.745265    2.913639    3.230656  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.000000   10.000000   10.000000   10.000000  
50%      2.000000   11.000000   11.000000   12.000000  
75%      6.000000   13.000000   13.000000   14.000000  
max     32.000000   19.000000   19.000000   19.000000

So we have 395 and 649 records for df1 and df2 respectively.

Now we can merge them,

In [13]:
df = pd.merge(df1, df2, on=["school","sex","age","address","famsize","Pstatus","Medu","Fedu","Mjob","Fjob","reason","nursery","internet"])

In [14]:
df.describe()

age        Medu        Fedu  traveltime_x  studytime_x  \
count  382.000000  382.000000  382.000000    382.000000   382.000000   
mean    16.586387    2.806283    2.565445      1.442408     2.034031   
std      1.173470    1.086381    1.096240      0.695378     0.845798   
min     15.000000    0.000000    0.000000      1.000000     1.000000   
25%     16.000000    2.000000    2.000000      1.000000     1.000000   
50%     17.000000    3.000000    3.000000      1.000000     2.000000   
75%     17.000000    4.000000    4.000000      2.000000     2.000000   
max     22.000000    4.000000    4.000000      4.000000     4.000000   

       failures_x    famrel_x  freetime_x     goout_x      Dalc_x  ...  \
count  382.000000  382.000000  382.000000  382.000000  382.000000  ...   
mean     0.290576    3.939791    3.222513    3.112565    1.473822  ...   
std      0.729481    0.921620    0.988233    1.131927    0.886229  ...   
min      0.000000    1.000000    1.000000    1.000000    1.000000  ...   
25%      0.000000    4.000000    3.000000    2.000000    1.000000  ...   
50%      0.000000    4.000000    3.000000    3.000000    1.000000  ...   
75%      0.000000    5.000000    4.000000    4.000000    2.000000  ...   
max      3.000000    5.000000    5.000000    5.000000    5.000000  ...   

         famrel_y  freetime_y     goout_y      Dalc_y      Walc_y    health_y  \
count  382.000000  382.000000  382.000000  382.000000  382.000000  382.000000   
mean     3.942408    3.230366    3.117801    1.476440    2.290576    3.575916   
std      0.908884    0.985096    1.133710    0.886303    1.282577    1.404248   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    3.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

       absences_y        G1_y        G2_y        G3_y  
count  382.000000  382.000000  382.000000  382.000000  
mean     3.672775   12.112565   12.238220   12.515707  
std      4.905965    2.556531    2.468341    2.945438  
min      0.000000    0.000000    5.000000    0.000000  
25%      0.000000   10.000000   11.000000   11.000000  
50%      2.000000   12.000000   12.000000   13.000000  
75%      6.000000   14.000000   14.000000   14.000000  
max     32.000000   19.000000   19.000000   19.000000  

[8 rows x 29 columns]

We have 382 students in both datasets.

Save back to file